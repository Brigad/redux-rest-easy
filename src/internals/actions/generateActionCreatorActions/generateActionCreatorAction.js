const generateActionCreatorAction = actionType => (
  normalizedURL,
  resourceId,
  normalizedPayload,
  principalResourceIds,
) => ({
  type: actionType,
  url: normalizedURL,
  resourceId,
  payload:
    normalizedPayload && Object.keys(normalizedPayload).length
      ? normalizedPayload
      : undefined,
  principalResourceIds:
    typeof principalResourceIds === 'string'
      ? [principalResourceIds]
      : principalResourceIds,
});

export default generateActionCreatorAction;
