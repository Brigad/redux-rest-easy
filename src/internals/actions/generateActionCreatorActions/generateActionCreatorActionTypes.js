import generateActionType from '../../utils/generateActionType';

const generateActionCreatorActionTypes = (resourceName, actionName) => ({
  REQUEST: generateActionType(resourceName, actionName, 'REQUEST'),
  RECEIVE: generateActionType(resourceName, actionName, 'RECEIVE'),
  FAIL: generateActionType(resourceName, actionName, 'FAIL'),
  RECEIVE_FROM_CACHE: generateActionType(
    resourceName,
    actionName,
    'RECEIVE_FROM_CACHE',
  ),
});

export default generateActionCreatorActionTypes;
