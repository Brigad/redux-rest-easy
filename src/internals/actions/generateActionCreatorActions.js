import generateActionCreatorAction from './generateActionCreatorActions/generateActionCreatorAction';
import generateActionCreatorActionTypes from './generateActionCreatorActions/generateActionCreatorActionTypes';

const generateActionCreatorActions = (resourceName, actionName) => {
  const {
    REQUEST,
    RECEIVE,
    FAIL,
    RECEIVE_FROM_CACHE,
  } = generateActionCreatorActionTypes(resourceName, actionName);

  return {
    REQUEST: generateActionCreatorAction(REQUEST),
    RECEIVE: generateActionCreatorAction(RECEIVE),
    FAIL: generateActionCreatorAction(FAIL),
    RECEIVE_FROM_CACHE: generateActionCreatorAction(RECEIVE_FROM_CACHE),
  };
};

export default generateActionCreatorActions;
