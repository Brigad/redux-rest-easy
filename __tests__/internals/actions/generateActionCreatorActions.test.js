import generateActionCreatorActions from '../../../src/internals/actions/generateActionCreatorActions';

describe('generateActionCreatorActions', () => {
  test('only path', () => {
    expect(generateActionCreatorActions()).toMatchSnapshot();
  });
});
