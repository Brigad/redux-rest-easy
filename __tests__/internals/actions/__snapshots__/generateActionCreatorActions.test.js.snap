// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateActionCreatorActions only path 1`] = `
Object {
  "FAIL": [Function],
  "RECEIVE": [Function],
  "RECEIVE_FROM_CACHE": [Function],
  "REQUEST": [Function],
}
`;
